### DISCUSSION TITLE
Integrate a prose linter

### DISCUSSION THREAD

**Discussion by endorama** [standardnotes/forum#2705] [Suggestions]
Hello Standard Notes team!

I'm an avid user of this app and an Extended subscriber :) I like the product and the concept.

But I'm not a native English speaker so I really benefit from prose linting. Spell checking (BTW thanks for having added it) is not enough to ensure beautiful writing.
Prose linting may also be valuable for improving language inclusiveness.

I was wondering how to include some prose linters, like [btford/write-good](https://github.com/btford/write-good) or [get-ales/alex](https://github.com/get-alex/alex). 

They would be an awesome addition to the app. I tried looking at the developer docs to understand how to integrate those, but I'm not sure how to proceed, as linting output should be shown somewhere and I've not seen examples of such behaviour in current extensions.

Do you have any hint or suggestion? Thanks!

**Comment by JaspalSuri** [comment]
I'll pass along your feature request and have assigned Baptiste to this so he can share some insight as to what might work. :)

**Comment by arielsvg** [comment]
That's a pretty cool suggestion; I'll move this to our forum so people can join in on the discussion!
There's not much you could right now aside from integrating such libraries into an editor.

**Comment by endorama** [comment]
Thanks for considering this!

A new editor seems a okish approach (I fear the duplication), unless you are considering having inline recommendations, or a "message window", where to display such linter's messages independently from the editor of choice.  

Do you think you would consider such an addition?

**Comment by danieljhill** [comment]
This would be really handy as someone who is writing prose using SN a lot of the time, especially in conjunction with something like https://github.com/amperser/proselint
